core/src/main/scala/org/apache/spark/rdd/PipedRDD.scala

```diff
@@ -147,7 +148,7 @@ private[spark] class PipedRDD[T: ClassTag](
       override def run(): Unit = {
         TaskContext.setTaskContext(context)
         val out = new PrintWriter(new BufferedWriter(
-          new OutputStreamWriter(proc.getOutputStream), bufferSize))
+          new OutputStreamWriter(proc.getOutputStream, StandardCharsets.UTF_8), bufferSize))
```

Comment:
This is the std out from another process. It should use the system default encoding, otherwise it may not work.